src/routes/build.svelte

```diff
@@ -88,17 +89,23 @@
     collectLog(buildLog, deployLog)
   })
 
-  onMount(() => {
+  onMount(async () => {
     // no repo url? Redirect back
     if (buildId === '') {
       showBuildError('"build_id" is missing')
     } else {
       // start fetching the log
-      build.get(buildId, true)
+      websocket = await build.get(buildId, true)
     }
   })
 
-  onDestroy(unsubscribe)
+  onDestroy(() => {
+    if (websocket) {
+      websocket.close()
+    }
```

Comment:
Suggestion: maybe you move this operation to be attached to `unsubscribe` itself, so in the store you'll do something like:

```js
return {
  subscribe: () => {
    const unsubscribe = subscribe()
    return () => {
      if (ws) { ws.close() } 
      unsubscribe()
    }
  },

Reply:
Good idea! 👍